enhancements/cluster-logging/fluentd-tweaks.md

```diff
@@ -0,0 +1,166 @@
+  E.g. a syslog output with tweaks could have a `syslog` *destination* section and `fluentd` tweaks section.
+
+## Summary
+
+Allow selected fields from the fluentd `<buffer>` configuration section to be
+specified for logging, for performance-tuning purposes. These settings are
+referred to as `tweaks` to clearly identify that they are for advanced
+performance-tuning use only.
+
+Tweaks are:
+* *not* relevant to most users, default settings should give good general performance.
+* only for *advanced* users with detailed knowledge of `fluentd` configuration and performance.
+* only for performance tuning, they have no effect on functional aspects of logging.
+
+Tweaks have weaker forward-compatibility guarantees than the rest of the API.
+In future releases:
+  - `fluentd` tweaks *may not apply* if the collector implementation changes.
+  - values set in tweaks *may be overridden* by settings in the collector or forwarder APIs.
+  - future collectors may or may not have a tweaks section
+  - future collectors may not have equivalent tweaks to `fluentd`.
+
+This will be presented in the `collectors.logging.openshift.io` API as a
+`tweaks` map field containing a `fluentd` map field. We may add other tweak
+types in future.
+
+
+## Motivation
+
+### Goals
+
+Ultimately we want great performance "out of the box" with no user
+intervention. However, today we can't always predict/detect the best settings;
+customers have had to adjust `fluentd` parameters to get good performance.
+
+Furthermore, some requirements *can't* be predicted or detected.  Optimizing for
+low latency often conflicts with optimizing for high throughput. We can't
+predict or detect that a user wants to aggressively optimize for one or the
+other.
+
+Goals:
+* Expose selected fluentd performance optimization parameters in the Collector API.
+* Apply user settings to the collectors `fluentd` configuration.
+
+We will be successful when:
+* An administrator is able to tune selected `fluentd` parameters for their own use case.
+
+### Non-Goals
+
+* Not implying the collector will always be `fluentd`. These are limited tweaks to optimize the existing collector, they may not apply in future.
+* Not a general `fluentd` configuration: we will only expose a minimal set of of performance tuning parameters. Users that want full control of `fluentd` need to go unmanaged.
+* Not generic performance tuning for collectors other than `fluentd`. Future collectors may or may not have their own `tweaks`
+
+## Proposal
+
+Field names are based on fluentd 1.0 `<buffer>` configuration section.
+See the [fluentd documentation](https://docs.fluentd.org/configuration/buffer-section#buffering-parameters) for details.
+
+Example below shows just the `tweaks` section of a Collector CR YAML file.
+Some field names changed between `fluentd` 0.12 and 1.0, noted in comments.
+
+```yaml
+tweaks:
+  fluentd:
+    buffer:
+      # Memory use
+      chunk_limit_size: 1m
```

Comment:
I like underscores but this is generally a ruby convention.  Are dashes or camelcase more traditional with yaml or go?

Reply:
Yes, in general. In this case I thought it best to copy the names exactly from fluentd. Only folks familiar with fluentd config will come here, and these names will be familiar. Also we're not adding any extra meaning or doing any sanity checks - we're just dropping this into our fluentd config, and assuming the user knows what that means. So in some ways the contrasting style is a good thing - this is different from our normal API.